control-plane-operator/controllers/hostedcontrolplane/csi/kubevirt/kubevirt.go

```diff
@@ -0,0 +1,386 @@
+	}
+
+	for i, container := range ds.Spec.Template.Spec.Containers {
+		switch container.Name {
+		case "csi-driver":
+			ds.Spec.Template.Spec.Containers[i].Image = csiDriverImage
+		case "csi-node-driver-registrar":
+			ds.Spec.Template.Spec.Containers[i].Image = csiNodeDriverRegistrarImage
+		case "csi-liveness-probe":
+			ds.Spec.Template.Spec.Containers[i].Image = csiLivenessProbeImage
+		}
+	}
+
+	return nil
+}
+
+func ReconcileTenant(client crclient.Client, hcp *hyperv1.HostedControlPlane, ctx context.Context, createOrUpdate upsert.CreateOrUpdateFN, componentImages map[string]string) error {
+
+	tenantNamespace := manifests.KubevirtCSIDriverTenantNamespaceStr
+
+	ns := manifests.KubevirtCSIDriverTenantNamespace(tenantNamespace)
+	_, err := createOrUpdate(ctx, client, ns, func() error { return nil })
+	if err != nil {
+		return err
+	}
+
+	tenantNodeServiceAccount := manifests.KubevirtCSIDriverTenantNodeSA(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantNodeServiceAccount, func() error {
+		return reconcileTenantNodeSA(tenantNodeServiceAccount)
+	})
+	if err != nil {
+		return err
+	}
+
+	tenantNodeClusterRole := manifests.KubevirtCSIDriverTenantNodeClusterRole(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantNodeClusterRole, func() error {
+		return reconcileTenantNodeClusterRole(tenantNodeClusterRole)
+	})
+	if err != nil {
+		return err
+	}
+
+	tenantNodeClusterRoleBinding := manifests.KubevirtCSIDriverTenantNodeClusterRoleBinding(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantNodeClusterRoleBinding, func() error {
+		return reconcileTenantNodeClusterRoleBinding(tenantNodeClusterRoleBinding, tenantNamespace)
+	})
+	if err != nil {
+		return err
+	}
+
+	tenantControllerClusterRoleBinding := manifests.KubevirtCSIDriverTenantControllerClusterRoleBinding(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantControllerClusterRoleBinding, func() error {
+		return reconcileTenantControllerClusterRoleBinding(tenantControllerClusterRoleBinding, tenantNamespace)
+	})
+	if err != nil {
+		return err
+	}
+
+	tenantControllerClusterRole := manifests.KubevirtCSIDriverTenantControllerClusterRole(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantControllerClusterRole, func() error {
+		return reconcileTenantControllerClusterRole(tenantControllerClusterRole)
+	})
+	if err != nil {
+		return err
+	}
+
+	tenantControllerServiceAccount := manifests.KubevirtCSIDriverTenantControllerSA(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, tenantControllerServiceAccount, func() error {
+		return reconcileTenantControllerSA(tenantControllerServiceAccount)
+	})
+	if err != nil {
+		return err
+	}
+
+	daemonSet := manifests.KubevirtCSIDriverDaemonSet(tenantNamespace)
+	_, err = createOrUpdate(ctx, client, daemonSet, func() error {
+		return reconcileTenantDaemonset(daemonSet, componentImages)
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
```

Comment:
what is "infra" in this context? can we add a comment defining it?

Reply:
infra refered to the components deployed underlying management ocp's hosted control plane. i added a comment to clarify this.